### ISSUE TITLE
Ticking ascends for routes that are not listed.

### ISSUE THREAD

**Issue by Wymiatator** [theCrag/website#4150]
For pure progress tracking and CPR calculation purpose I would like to tick ascends that are not listed on theCrag.
Mostly used for gyms that don't add any routes or are not even listed.

**Comment by scd** [comment]
Not possible to tick routes that are not registered. It is fundamental to our philosophy and design. 

So where to here?

Gym routes, you can add the gym and their routes. This is something that we have been spending a lot of time supporting the workflow which is a little different from outdoor.

Some people are experimentally working on Moonboard/Kilterboard. But there are some teething issues with how this can be integrated into theCrag.

**Comment by Wymiatator** [comment]
I travel a lot and visit a lot of gyms around the world, most of them don't have any routes listed on theCrag or the gyms themselves are not listed at all. I'm not able to get permission to edit in every one of them just to tick couple climbs, not mentioning the time it would take me to fill all the data. I'd rather have some generic gym with generic routes just to track my progress, otherwise I will never be able to use the full potential of the CPR.

Please let me know if you know how I make this work in my case. Thanks in advance.

**Comment by scd** [comment]
Generic routes don't work with CPR as the system will think it is a repeat. This is a chicken or egg problem and the same one we had with outdoor routes.

Note that the full potential of CPR requires both distinct routes for distinct ascents and registered route grades registered by the community.

**Comment by Wymiatator** [comment]
Yeah that's why ideally I would like to have the possibility to log an ascent of a route that is not registered but exists in a gym, has an assigned grade by the gyms route setter, the height is known and all other variables. This would be some kind of a phantom route that perhaps only I can see on my profile.

**Comment by scd** [comment]
I love it that you want to log gym routes. I did this for a while with my local gym sending me a spreadsheet of their resets. But they wanted to see lot's of people ticking.

The main argument for me is that it is a stepping stone to get gyms to manage their routes on theCrag if people are already ticking their routes on theCrag. It is only a stepping stone though because the full solution of gyms managing resets on theCrag gives us a huge amount of features that we can work with.

Biggest problem here is that even if we come up with a good stepping stone solution for this we really don't have the development resources to do anything for a while.

By all means keep discussing it because either you, me or somebody else may come up with a really nice idea of how this may work.

BTW, most gyms I climb in currently have their own grading system. Something we can deal with if the gym owners tell us about the grading system and manage their own routes. Pretty difficult to deal with if you just climbed the Red route and there is no contextual information about what Red means,
